src/Microsoft.NetCore.Analyzers/Core/Runtime/DoNotUseEnumerableMethodsOnIndexableCollectionsInsteadUseTheCollectionDirectly.Fixer.cs

```diff
@@ -77,14 +79,48 @@ private async Task<Document> UseCollectionDirectly(Document document, TextSpan s
             }
 
             var generator = SyntaxGenerator.GetGenerator(document);
-            var indexNode = generator.LiteralExpression(0);
-            var elementAccessNode = generator.ElementAccessExpression(collectionSyntax.WithoutTrailingTrivia(), indexNode)
-                .WithTrailingTrivia(invocationNode.GetTrailingTrivia());
 
-            var newRoot = root.ReplaceNode(invocationNode, elementAccessNode);
+            var elementAccessNode = GetReplacementNode(generator, collectionSyntax, methodName);
+            if (elementAccessNode == null)
+            {
+                return document;
+            }
+
+            var newRoot = root.ReplaceNode(invocationNode, elementAccessNode.WithTrailingTrivia(invocationNode.GetTrailingTrivia()));
             return document.WithSyntaxRoot(newRoot);
         }
 
+        private static SyntaxNode GetReplacementNode(SyntaxGenerator generator, SyntaxNode collectionSyntax, string methodName)
+        {
+            var collectionSyntaxNoTrailingTrivia = collectionSyntax.WithoutTrailingTrivia();
+
+            if (methodName == "First")
+            {
+                var zeroLiteral = generator.LiteralExpression(0);
+                return generator.ElementAccessExpression(collectionSyntaxNoTrailingTrivia, zeroLiteral);
+            }
+
+            if (methodName == "Last")
+            {
+                // TODO: Handle C# 8 index expression (and vb.net equivalent if any)
+                // TODO: Handle cases were `collectionSyntax` is an invocation. We would need to create some intermediate variable.
```

Comment:
I haven't handled this case yet because it adds quite a lot of complexity. I will need to find the previous statement to insert a new variable, find an available variable name and in the case of arrow expression I could need to change the arrow expression into a block.

I am fine to work on this but I might need some help.

Reply:
I think it is fine to not do it for now.

Reply:
Shall I remove the todo comment? Shall I create some separate ticket?

Reply:
Obviously `FirstOrDefault` and `LastOrDefault` will fall under the same need.